app/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java

```diff
@@ -413,91 +415,103 @@ private HibernateAccount getHibernateAccount(AccountId accountId) {
-                parameters.put("regionCode", unguarded.getPhone().getRegionCode());
+                builder.append("AND acct.phone.number=:number AND acct.phone.regionCode=:regionCode",
+                        "number", unguarded.getPhone().getNumber(),
+                        "regionCode", unguarded.getPhone().getRegionCode());
             } else {
-                query = EXTID_QUERY;
-                parameters.put("studyId", unguarded.getStudyId());
-                parameters.put("externalId", unguarded.getExternalId());
+                builder.append("AND acct.externalId=:externalId", "externalId", unguarded.getExternalId());
+            }
+        }
+        if (search != null) {
+            if (StringUtils.isNotBlank(search.getEmailFilter())) {
+                builder.append("AND acct.email LIKE :email", "email", "%"+search.getEmailFilter()+"%");
+            }
+            if (StringUtils.isNotBlank(search.getPhoneFilter())) {
+                String phoneString = search.getPhoneFilter().replaceAll("\\D*", "");
+                builder.append("AND acct.phone.number LIKE :number", "number", "%"+phoneString+"%");
             }
-            List<HibernateAccount> accountList = hibernateHelper.queryGet(query, parameters, null, null, HibernateAccount.class);
-            if (accountList.isEmpty()) {
-                return null;
+            if (search.getStartTime() != null) {
+                builder.append("AND acct.createdOn >= :startTime", "startTime", search.getStartTime());
             }
-            hibernateAccount = accountList.get(0);
-            if (accountList.size() > 1) {
-                LOG.warn("Multiple accounts found email/phone query; example accountId=" + hibernateAccount.getId());
+            if (search.getEndTime() != null) {
+                builder.append("AND acct.createdOn <= :endTime", "endTime", search.getEndTime());
             }
+            if (search.getLanguage() != null) {
+                builder.append("AND :language IN ELEMENTS(acct.languages)", "language", search.getLanguage());
+            }
+            builder.dataGroups(search.getAllOfGroups(), "IN");
+            builder.dataGroups(search.getNoneOfGroups(), "NOT IN");
         }
-        return hibernateAccount;
-    }
-
-    /** {@inheritDoc} */
-    @Override
-    public void deleteAccount(AccountId accountId) {
-        String userId = accountId.getUnguardedAccountId().getId();
-        if (userId == null) {
-            HibernateAccount hibernateAccount = getHibernateAccount(accountId);
-            userId = hibernateAccount.getId();
+        Set<String> callerSubstudies = context.getCallerSubstudies();
+        if (!callerSubstudies.isEmpty()) {
+            builder.append("AND acctSubstudy.substudyId IN (:substudies)", "substudies", callerSubstudies);
         }
-        hibernateHelper.deleteById(HibernateAccount.class, userId);    
+        if (!isCount) {
+            builder.append("GROUP BY acct.id");        
+        }
```

Comment:
Because this query has the HibernateAccount() constructor in it, substudies are not retrieved at all. You cannot include a collection in the constructor apparently (it's not supported by JPA), and I didn't find a way around this (except to load the whole record perhaps, but since the record could include a large blob of JSON, I'm still avoiding that for the summary query). For the query of an individual record by an identifier... the GROUP BY clause doesn't do anything as there's only one account ID